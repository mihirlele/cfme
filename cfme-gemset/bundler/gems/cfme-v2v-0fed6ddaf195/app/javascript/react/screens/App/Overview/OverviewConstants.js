// reminder... we need to go back and prefix these: V2V_
export const SHOW_PLAN_WIZARD = 'SHOW_PLAN_WIZARD';
export const SHOW_PLAN_WIZARD_EDIT_MODE = 'SHOW_PLAN_WIZARD_EDIT_MODE';
export const HIDE_PLAN_WIZARD = 'HIDE_PLAN_WIZARD';
export const SHOW_EDIT_PLAN_TITLE_MODAL = 'SHOW_EDIT_PLAN_TITLE_MODAL';
export const HIDE_EDIT_PLAN_TITLE_MODAL = 'HIDE_EDIT_PLAN_TITLE_MODAL';
export const PLAN_WIZARD_EXITED = 'PLAN_WIZARD_EXITED';
export const PLAN_WIZARD_NEXT = 'PLAN_WIZARD_NEXT';
export const PLAN_WIZARD_BACK = 'PLAN_WIZARD_BACK';
export const CONTINUE_TO_PLAN = 'CONTINUE_TO_PLAN';
export const V2V_SET_MIGRATIONS_FILTER = 'V2V_SET_MIGRATIONS_FILTER';
export const V2V_AUTO_SET_MIGRATIONS_FILTER = 'V2V_AUTO_SET_MIGRATIONS_FILTER';
export const V2V_RETRY_MIGRATION = 'V2V_RETRY_MIGRATION';
export const V2V_TOGGLE_SCHEDULE_MIGRATION_MODAL = 'V2V_TOGGLE_SCHEDULE_MIGRATION_MODAL';
export const V2V_SCHEDULE_MIGRATION = 'V2V_SCHEDULE_MIGRATION';
export const V2V_EDIT_PLAN_REQUEST = 'V2V_EDIT_PLAN_REQUEST';
export const V2V_CANCEL_PLAN_REQUEST = 'V2V_CANCEL_PLAN_REQUEST';
export const SHOW_CONFIRM_MODAL = 'SHOW_CONFIRM_MODAL';
export const HIDE_CONFIRM_MODAL = 'HIDE_CONFIRM_MODAL';
export const ARCHIVE_TRANSFORMATION_PLAN = 'ARCHIVE_TRANSFORMATION_PLAN';
export const DELETE_TRANSFORMATION_PLAN = 'DELETE_TRANSFORMATION_PLAN';

export const FETCH_V2V_TRANSFORMATION_MAPPINGS = 'FETCH_V2V_TRANSFORMATION_MAPPINGS';

export const FETCH_V2V_TRANSFORMATION_PLANS = 'FETCH_V2V_TRANSFORMATION_PLANS';
export const FETCH_V2V_SERVICE_TEMPLATE_ANSIBLE_PLAYBOOKS = 'FETCH_V2V_SERVICE_TEMPLATE_ANSIBLE_PLAYBOOKS';
export const FETCH_V2V_ARCHIVED_TRANSFORMATION_PLANS = 'FETCH_V2V_ARCHIVED_TRANSFORMATION_PLANS';

export const CREATE_V2V_TRANSFORMATION_PLAN_REQUEST = 'CREATE_V2V_TRANSFORMATION_PLAN_REQUEST';

export const FETCH_V2V_ALL_REQUESTS_WITH_TASKS = 'FETCH_V2V_ALL_REQUESTS_WITH_TASKS';
export const FETCH_V2V_ALL_ARCHIVED_PLAN_REQUESTS_WITH_TASKS = 'FETCH_V2V_ALL_ARCHIVED_PLAN_REQUESTS_WITH_TASKS';

export const MIGRATIONS_FILTERS = {
  notStarted: __('Not Started Plans'),
  inProgress: __('In Progress Plans'),
  completed: __('Completed Plans'),
  archived: __('Archived Plans')
};

export const OSP_TENANT = 'CloudTenant';
export const OSP_VOLUME = 'CloudVolumeType';
export const OSP_NETWORK = 'CloudNetwork';
export const EMS_CLUSTER = 'EmsCluster';

export const FETCH_TRANSFORMATION_PLANS_URL =
  '/api/service_templates/?' +
  "filter[]=type='ServiceTemplateTransformationPlan'" +
  '&filter[]=active=true' +
  '&expand=resources,schedules' +
  '&attributes=name,description,miq_requests,options,created_at,transformation_mapping,transformation_mapping.transformation_mapping_items' +
  '&sort_by=updated_at' +
  '&sort_order=desc';

export const FETCH_ARCHIVED_TRANSFORMATION_PLANS_URL =
  '/api/service_templates/?' +
  "filter[]=type='ServiceTemplateTransformationPlan'" +
  '&filter[]=archived=true' +
  '&expand=resources' +
  '&attributes=name,description,miq_requests,options,created_at,transformation_mapping' +
  '&sort_by=updated_at' +
  '&sort_order=desc';

export const TRANSFORMATION_PLAN_REQUESTS_URL = '/api/requests';

export const WAITING_FOR_CONVERSION_HOST_MESSAGES = {
  notAvailable: __('Waiting for an available conversion host. You can continue waiting or go to the Migration Settings page to increase the number of migrations per host'), // prettier-ignore
  noRsaKey: __('No RSA key pair exists on the OpenStack provider. Cancel the migration, add the key, and retry.')
};
