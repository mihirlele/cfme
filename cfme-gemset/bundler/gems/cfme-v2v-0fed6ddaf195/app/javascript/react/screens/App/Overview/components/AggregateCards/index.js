import NotStartedTransformationPlans from './NotStartedTransformationPlans/NotStartedTransformationPlans';
import ActiveTransformationPlans from './ActiveTransformationPlans/ActiveTransformationPlans';
import FinishedTransformationPlans from './FinishedTransformationPlans/FinishedTransformationPlans';
import ArchivedTransformationPlans from './ArchivedTransformationPlans/ArchivedTransformationPlans';

export {
  NotStartedTransformationPlans,
  ActiveTransformationPlans,
  FinishedTransformationPlans,
  ArchivedTransformationPlans
};
