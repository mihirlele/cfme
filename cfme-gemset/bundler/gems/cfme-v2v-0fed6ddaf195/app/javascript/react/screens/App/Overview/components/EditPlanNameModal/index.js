import { connect } from 'react-redux';
import EditPlanNameModal from './EditPlanNameModal';
import { findEditingPlan } from '../../screens/PlanWizard/PlanWizardSelectors';
import * as EditPlanNameActions from './EditPlanNameActions';

import reducer from './EditPlanNameReducer';

export const reducers = { editPlanName: reducer };

const mapStateToProps = ({ editPlanName, overview, form }) => {
  const plans = [...overview.transformationPlans, ...overview.archivedTransformationPlans];
  const editingPlan = findEditingPlan(plans, overview.editingPlanId);
  return {
    ...editPlanName,
    editPlanNameModal: form && form.editPlanNameModal,
    initialValues: {
      name: editingPlan ? editingPlan.name : '',
      description: editingPlan ? editingPlan.description : ''
    },
    enableReinitialize: true,
    editingPlan
  };
};

export default connect(
  mapStateToProps,
  EditPlanNameActions
)(EditPlanNameModal);
