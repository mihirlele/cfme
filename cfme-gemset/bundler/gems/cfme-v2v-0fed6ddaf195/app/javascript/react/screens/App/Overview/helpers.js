import getMostRecentRequest from '../common/getMostRecentRequest';
import { TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES } from '../Mappings/screens/MappingWizard/MappingWizardConstants';

export const planTransmutation = (plans = [], mappings = []) =>
  plans.map(plan => {
    const request = getMostRecentRequest(plan.miq_requests);
    return {
      ...plan,
      infraMappingName: plan.transformation_mapping && plan.transformation_mapping.name,
      status: request ? request.status : null,
      configVmLength: plan.options.config_info.actions.length,
      scheduleTime: plan.schedules ? new Date(plan.schedules[0].run_at.start_time).getTime() : null
    };
  });

export const attachTargetProvider = (plan, providers, clusters, targetProviderType) => {
  if (!plan.transformation_mapping) {
    return plan;
  }

  const {
    transformation_mapping: { transformation_mapping_items }
  } = plan;
  const clusterMapping = transformation_mapping_items.find(
    item => item.destination_type === TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES[targetProviderType].cluster
  );
  const targetCluster = clusters.find(cluster => cluster.id === clusterMapping.destination_id);
  if (targetCluster === undefined) {
    return plan;
  }
  const targetProvider = providers.find(provider => provider.id === targetCluster.ems_id);

  return { ...plan, targetProvider };
};
