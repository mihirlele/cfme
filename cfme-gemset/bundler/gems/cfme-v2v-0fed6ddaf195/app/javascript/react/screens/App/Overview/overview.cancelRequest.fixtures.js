import Immutable from 'seamless-immutable';

export const cancelRequestResponse = Immutable({
  href: 'http://0.0.0.0:8080/api/requests/28',
  message: 'MiqRequest 28 canceled',
  success: true
});
