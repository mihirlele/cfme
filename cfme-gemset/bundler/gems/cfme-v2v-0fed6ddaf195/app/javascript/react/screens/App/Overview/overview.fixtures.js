import Immutable from 'seamless-immutable';

export const initialState = Immutable({
  mappingWizardVisible: false,
  hideMappingWizard: false,
  planWizardVisible: false,
  hidePlanWizard: false,
  transformationMappings: [],
  isRejectedTransformationMappings: false,
  isFetchingTransformationMappings: false,
  fetchTransformationMappingsUrl: '/api/dummyMappings',
  transformationPlans: [],
  fetchTransformationPlansUrl: '/api/dummyTransformationPlans',
  fetchArchivedTransformationPlansUrl: '/api/dummyArchivedTransformationPlans',
  fetchServiceTemplateAnsiblePlaybooksUrl: '/api/dummyServiceTemplatePlaybooks',
  clusters: [],
  isRejectedClusters: false,
  isFetchingClusters: false,
  fetchClustersUrl: '/api/dummyClusters',
  datastores: [],
  isRejectedDatastores: false,
  isFetchingDatastores: false,
  fetchDatastoresUrl: '/api/dummyDatastores',
  networks: [],
  isRejectedNetworks: false,
  isFetchingNetworks: false,
  fetchNetworksUrl: '/api/dummyNetworks',
  archivedTransformationPlans: [],
  requestsProcessingCancellation: []
});

export const transformationMappings = [
  {
    id: '10000000000001',
    name: 'Mapping 1',
    description: 'infrastructre mapping 1 description',
    comments: null,
    state: 'draft',
    options: {},
    tenant_id: null,
    created_at: '2018-02-07T16:02:39Z',
    updated_at: '2018-02-07T16:02:39Z',
    transformation_mapping_items: [
      {
        id: '10000000000001',
        source_id: '10000000000002',
        source_type: 'EmsCluster',
        destination_id: '10000000000005',
        destination_type: 'EmsCluster',
        transformation_mapping_id: '10000000000001',
        options: {},
        created_at: '2018-02-07T16:01:44Z',
        updated_at: '2018-02-07T16:02:39Z'
      },
      {
        id: '10000000000002',
        source_id: '10000000000001',
        source_type: 'Storage',
        destination_id: '10000000000018',
        destination_type: 'Storage',
        transformation_mapping_id: '10000000000002',
        options: {},
        created_at: '2018-02-07T16:05:04Z',
        updated_at: '2018-02-07T16:05:04Z'
      }
    ]
  },
  {
    id: '10000000000002',
    name: 'Mapping 2',
    description: 'infrastructre mapping 2 description',
    comments: null,
    state: 'draft',
    options: {},
    tenant_id: null,
    created_at: '2018-02-07T16:02:39Z',
    updated_at: '2018-02-07T16:02:39Z',
    transformation_mapping_items: [
      {
        id: '10000000000001',
        source_id: '10000000000003',
        source_type: 'EmsCluster',
        destination_id: '10000000000005',
        destination_type: 'EmsCluster',
        transformation_mapping_id: '10000000000002',
        options: {},
        created_at: '2018-02-07T16:01:44Z',
        updated_at: '2018-02-07T16:02:39Z'
      },
      {
        id: '10000000000004',
        source_id: '10000000000004',
        source_type: 'Storage',
        destination_id: '10000000000019',
        destination_type: 'Storage',
        transformation_mapping_id: '10000000000002',
        options: {},
        created_at: '2018-02-07T16:05:04Z',
        updated_at: '2018-02-07T16:05:04Z'
      }
    ]
  }
];

export const createTransformationPlanRequestResponse = {
  href: 'http://localhost:3000/api/service_requests/30',
  id: '30',
  description: 'Provisioning Service [plan_test6] from [plan_test6]',
  approval_state: 'pending_approval',
  type: 'ServiceTemplateTransformationPlanRequest',
  created_on: '2018-02-22T22:50:41Z',
  updated_on: '2018-02-22T22:50:41Z',
  fulfilled_on: null,
  requester_id: '1',
  requester_name: 'Administrator',
  request_type: 'transformation_plan',
  request_state: 'pending',
  message: 'VM Transformations - Request Created',
  status: 'Ok',
  options: {
    dialog: null,
    workflow_settings: {
      resource_action_id: '312'
    },
    initiator: null,
    src_id: '32',
    cart_state: 'ordered',
    requester_group: 'EvmGroup-super_administrator'
  },
  userid: 'admin',
  source_id: '32',
  source_type: 'ServiceTemplate',
  destination_id: null,
  destination_type: null,
  tenant_id: '1',
  service_order_id: '41',
  process: true
};

export const requestTransformationMappingsData = {
  method: 'GET',
  fetchTransformationMappingsUrl: initialState.fetchTransformationMappingsUrl,
  response: { data: { resources: transformationMappings } }
};

export const createTransformationPlanRequestData = {
  method: 'POST',
  response: { data: createTransformationPlanRequestResponse }
};

export const preMigrationPlaybook = {
  href: 'http://localhost:3000/api/service_templates/43',
  id: '43',
  name: 'Ansible test',
  description: '',
  guid: '7de52447-2946-409d-8e76-b64d0f17803d',
  type: 'ServiceTemplateAnsiblePlaybook',
  service_template_id: null,
  options: {
    config_info: {
      provision: {
        repository_id: '23',
        playbook_id: '309',
        credential_id: '10',
        hosts: 'test_avaleror',
        verbosity: '1',
        log_output: 'on_error',
        extra_vars: {},
        execution_ttl: '5',
        become_enabled: false,
        cloud_credential_id: '124',
        new_dialog_name: 'demo_httpd',
        fqname: '/Service/Generic/StateMachines/GenericLifecycle/provision',
        dialog_id: '50'
      },
      retirement: {
        remove_resources: 'no_with_playbook',
        verbosity: '0',
        log_output: 'on_error',
        repository_id: '23',
        playbook_id: '305',
        credential_id: '10',
        execution_ttl: '',
        hosts: 'localhost',
        extra_vars: {},
        become_enabled: false,
        fqname: '/Service/Generic/StateMachines/GenericLifecycle/Retire_Advanced_Resource_None'
      }
    }
  },
  created_at: '2018-02-09T19:33:42Z',
  updated_at: '2018-02-11T12:31:20Z',
  display: true,
  evm_owner_id: null,
  miq_group_id: '39',
  service_type: 'atomic',
  prov_type: 'generic_ansible_playbook',
  provision_cost: null,
  service_template_catalog_id: '16',
  long_description: null,
  tenant_id: '1',
  generic_subtype: null,
  deleted_on: null
};
