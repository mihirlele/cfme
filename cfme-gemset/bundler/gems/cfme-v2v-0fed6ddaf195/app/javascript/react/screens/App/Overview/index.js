import { connect } from 'react-redux';
import Overview from './Overview';
import * as OverviewActions from './OverviewActions';
import * as NotificationActions from '../common/NotificationList/NotificationListActions';
import * as ProvidersActions from '../../../../redux/common/providers/providersActions';
import * as RouterActions from '../../../../redux/actions/routerActions';

import reducer from './OverviewReducer';
import {
  notStartedTransformationPlansFilter,
  activeTransformationPlansFilter,
  finishedTransformationPlansFilter,
  requestsProcessingCancellationFilter,
  combineRequestsProcessingCancellation,
  attachTargetProviderToOspPlans
} from './OverviewSelectors';

export const reducers = { overview: reducer, form: {} };

const mapStateToProps = (
  { overview, overview: { transformationPlans, allRequestsWithTasks, planId, cloudTenants }, providers },
  ownProps
) => {
  const { requestsProcessingCancellation, ...overviewRest } = overview;

  return {
    ...overviewRest,
    ...providers,
    ...ownProps.data,
    notStartedTransformationPlans: notStartedTransformationPlansFilter(transformationPlans),
    activeTransformationPlans: attachTargetProviderToOspPlans(
      activeTransformationPlansFilter(transformationPlans, planId),
      providers.providers,
      cloudTenants
    ),
    finishedTransformationPlans: finishedTransformationPlansFilter(transformationPlans),
    requestsProcessingCancellation: combineRequestsProcessingCancellation(
      requestsProcessingCancellation,
      requestsProcessingCancellationFilter(transformationPlans)
    )
  };
};

const mergeProps = (stateProps, dispatchProps, ownProps) => Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(
  mapStateToProps,
  Object.assign(OverviewActions, NotificationActions, ProvidersActions, RouterActions),
  mergeProps
)(Overview);
