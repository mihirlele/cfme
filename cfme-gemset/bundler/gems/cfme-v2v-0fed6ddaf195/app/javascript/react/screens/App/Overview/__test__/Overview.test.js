import React from 'react';
import { shallow } from 'enzyme';

import Overview from '../Overview';
import { coreComponents } from '../../../../../components';
import componentRegistry from '../../../../../components/componentRegistry';

jest.mock('../../../../../components/componentRegistry');
jest.useFakeTimers();

componentRegistry.registerMultiple(coreComponents);

describe('Overview component', () => {
  const getBaseProps = () => ({
    store: {},
    planWizardVisible: false,
    isFetchingTransformationMappings: false,
    isRejectedTransformationMappings: false,
    transformationMappings: [],
    transformationPlans: [],
    fetchTransformationPlansUrl: '',
    fetchServiceTemplateAnsiblePlaybooksUrl: '',
    archivedTransformationPlans: [],
    showPlanWizardAction: jest.fn(),
    fetchProvidersAction: jest.fn().mockReturnValue(Promise.resolve()),
    fetchTransformationMappingsAction: jest.fn(),
    fetchTransformationPlansAction: jest.fn().mockReturnValue(Promise.resolve()),
    fetchServiceTemplateAnsiblePlaybooksAction: jest.fn().mockReturnValue(Promise.resolve()),
    fetchCloudTenantsAction: jest.fn(),
    redirectTo: jest.fn()
  });

  describe('polling', () => {
    test('starts when the component is mounted', async () => {
      const props = getBaseProps();
      const wrapper = await shallow(<Overview {...props} />); // eslint-disable-line no-unused-vars
      expect(props.fetchTransformationPlansAction).toHaveBeenCalled();
    });

    test('fetches transformation plan requests every 15 seconds', async () => {
      const props = getBaseProps();
      const wrapper = await shallow(<Overview {...props} />); // eslint-disable-line no-unused-vars
      jest.advanceTimersByTime(15000);
      expect(props.fetchTransformationPlansAction).toHaveBeenCalledTimes(2);
    });
  });

  describe('checks for necessary providers', () => {
    test('and displays empty state if insufficient', () => {
      const props = getBaseProps();
      const wrapper = shallow(<Overview {...props} hasSufficientProviders={false} />); // eslint-disable-line no-unused-vars
      expect(wrapper.find('ShowWizardEmptyState')).toMatchSnapshot();
    });
  });
});
