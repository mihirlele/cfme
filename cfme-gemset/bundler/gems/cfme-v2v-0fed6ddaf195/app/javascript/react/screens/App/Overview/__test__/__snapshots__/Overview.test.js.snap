// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview component checks for necessary providers and displays empty state if insufficient 1`] = `
<ShowWizardEmptyState
  buttonHref="/ems_infra/show_list"
  buttonText="Configure Providers"
  className="full-page-empty"
  description="The source and target providers must be configured before attempting a migration"
  iconName="add-circle-o"
  iconType="pf"
  showWizardAction={null}
  title=" "
/>
`;
